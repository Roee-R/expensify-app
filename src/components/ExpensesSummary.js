import React from 'react';
import numeral from 'numeral';

const ExpensesSummary = (props)=>{
    const size = props.size;
    const sum=numeral(props.sum/100).format('$0,0.00');
    return (
        <p>Viewing {size} {size>1? 'expenses ' : 'expense '}totalling {sum}
        </p> 
    )
}

export default ExpensesSummary
