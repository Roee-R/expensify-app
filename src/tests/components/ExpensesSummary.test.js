import React from 'react';
import { shallow } from 'enzyme';
import numeral from 'numeral';

import ExpensesSummary from '../../components/ExpensesSummary';

let sum,size;

describe('ExpensesSummary tests', ()=>{
    test('should retrun summary of several expenses',()=>{
        const wraper = shallow(<ExpensesSummary 
            size={3}
            sum={111}
            />);
        sum=numeral(111).format('$0,0.00');  
        const returnedText = `Viewing 3 expenses totalling $1.11`          
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').text()).toBe(returnedText);
    })

    test('should retrun summary of one expense',()=>{    
        const wraper = shallow(<ExpensesSummary 
            size={1}
            sum={11}
            />);
        sum=numeral(11).format('$0,0.00');            
        const returnedText = `Viewing 1 expense totalling $0.11`    
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').text()).toBe(returnedText);
    })
})