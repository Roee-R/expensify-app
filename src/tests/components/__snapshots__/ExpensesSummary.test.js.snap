// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary tests should retrun summary of one expense 1`] = `
<p>
  Viewing 
  1
   
  expense 
  totalling 
  $0.11
</p>
`;

exports[`ExpensesSummary tests should retrun summary of several expenses 1`] = `
<p>
  Viewing 
  3
   
  expenses 
  totalling 
  $1.11
</p>
`;
